go/oasis-test-runner/oasis/validator.go

```diff
@@ -107,25 +123,45 @@ func (net *Network) NewValidator(cfg *ValidatorCfg) (*Validator, error) {
 		return nil, fmt.Errorf("oasis/validator: failed to provision validator identity: %w", err)
 	}
 
+	valPublicKey := valIdentity.NodeSigner.Public()
+
 	val := &Validator{
 		net:           net,
 		dir:           valDir,
 		entity:        cfg.Entity,
+		sentryIndices: cfg.SentryIndices,
+		tmAddress:     crypto.PublicKeyToTendermint(&valPublicKey).Address().String(),
 		consensusPort: net.nextNodePort,
 		grpcDebugPort: net.nextNodePort + 1,
 	}
 
-	valConsensusAddr := node.ConsensusAddress{
-		ID: valIdentity.NodeSigner.Public(),
-	}
-	if err = valConsensusAddr.Address.FromIP(netPkg.ParseIP("127.0.0.1"), val.consensusPort); err != nil {
-		return nil, fmt.Errorf("oasis/validator: failed to parse IP: %w", err)
+	var consensusAddress node.ConsensusAddress
+	localhost := netPkg.ParseIP("127.0.0.1")
+	if len(val.sentryIndices) > 0 {
+		// Use the first sentry node's public address for initial validator
+		// registration.
+		// NOTE: This only applies to the Genesis document and the first epoch.
```

Comment:
Any reason why we don't update the `registry node init` to allow specifying multiple consensus addresses?

Reply:
Great suggestion, I've done that.